src/Http/Http.Abstractions/src/Routing/EndpointMetadataCollection.cs

```diff
@@ -126,6 +126,20 @@ public EndpointMetadataCollection(params object[] items)
         return results;
     }
 
+    /// <summary>
+    /// Gets the most significant metadata item of type <typeparamref name="T"/>.
+    /// Throws an <see cref="InvalidOperationException"/> if the metadata is not found.
+    /// </summary>
+    /// <typeparam name="T">The type of metadata to retrieve.</typeparam>
+    /// <returns>
+    /// The most significant metadata of type <typeparamref name="T"/>.
+    /// </returns>
+    public T GetRequiredMetadata<T>() where T : class
```

Comment:
```suggestion
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public T GetRequiredMetadata<T>() where T : class
```
Does this method also need `AggressiveInlining` like the other one?
https://github.com/dotnet/aspnetcore/blob/0bc20453b290fa72a913fe4744cdcdff64663a1e/src/Http/Http.Abstractions/src/Routing/EndpointMetadataCollection.cs#L72-L73

Reply:
AFAIK throwing an exception prevents a method from being inlined.